src/utilities/extendSchema.js

```diff
@@ -257,14 +267,24 @@ export function extendSchema(
   // this scope and have access to the schema, cache, and newly defined types.
 
   function getMergedDirectives(): Array<GraphQLDirective> {
-    const existingDirectives = schema.getDirectives();
+    const existingDirectives = schema.getDirectives().map(extendDirectiveType);
     invariant(existingDirectives, 'schema must have default directives');
 
     return existingDirectives.concat(
       directiveDefinitions.map(node => astBuilder.buildDirective(node)),
     );
   }
 
+  function extendDirectiveType(type: GraphQLDirective): GraphQLDirective {
```

Comment:
function name is confusing it should be either `extendDirective` or `extendDirectiveTypes` (I like it more).

Reply:
I'm going with `extendDirective`